### PR TITLE
FLOE-576: overlapping table of contents at line spacing less than 1

### LINKED ISSUE #110: Contents of "Table of Contents" overlap with each other when line spacing is less than 1.
## Describe the bug
Contents of "Table of Contents" overlap with each other when line spacing is less than 1.

### To reproduce
1. Go to Preferences.
2. Turn on the table of contents.
3. Reduce the line spacing to some number less than 0.9

### Expected behavior
The different points in the table of contents should not overlap when line spacing is decreased. 

## Screenshots
87943276_565176140759600_2311083350351151104_n

### Desktop
 - OS: Ubuntu 18.04
 - Browser: Chrome 
 - Version: 80.0.3987.122 (Official Build) (64-bit)

## Additional context or notes
A temporary solution to this problem would be to place a cap on the minimum value of line spacing.

### PR DESCRIPTION
**Pull request by sachin10101998**
Fixes #110 
Screenshot:
Screenshot from 2020-02-28 01-39-02

### CODE REVIEW

**Comment by aashimgarg** [comment]
I think that does not serve the purpose as you are adding the margin properties in the text when the line spacing is 1 . Rather when we decrease the line space the text should be responsive and they should not collide and moreover the space between them should also decrease . And if you add margin properties when the line space is 1 the line space is actually not 1 because the void space is actually covered by margin not line space.
What are your thoughts @jobara @sachin10101998 ?

**Comment by sachin10101998** [comment]
@aashimgarg , it's working for all values of line spacing and not just when line spacing is 1. 
aaa

**Comment by aashimgarg** [comment]
You did not get my point , by adding the margin properties to the table of contents it actually increases the line space between them for every case.
for example in case you selected the line space to be .8 , so in that case the line space is actually not .8 it is more than that as you added margin bottom and margin top to the table of contents .
The text does not collide or overlap with each other because you gave the contents that much margin so that the contents does not not collide when the line space between them is least , in this case which is .7 .

**Comment by sachin10101998** [comment]
> The text does not collide or overlap with each other because you gave the contents that much margin so that the contents does not not collide when the line space between them is least , in this case which is .7 .

But isn't that exactly what a solution to this issue is supposed to be?

>in case you selected the line space to be .8 , so in that case the line space is actually not .8 it is more than that as you added margin bottom and margin top to the table of contents

I am not sure what units were used while setting up the line spacing values in the UI options. But the 'line-height' property changes as expected in relation to the set value of line spacing and keeps a check on the minimum to prevent overlapping of contents.

**Comment by jobara** [comment]
@sachin10101998 and @aashimgarg thanks for discussing this. The purpose of adjusting the line-height is less about setting an absolute value and more about adjusting the presentation of the content to something that suits your needs and preferences. It's unlikely that you'd want to shrink the line-height to a point where the text overlaps. However, if the default styling is to have the text wide apart then you may want to tighten it up. That being said, we want good defaults, so the "1" case should not look stretched.

File: css/style.css [hunk 1]
```diff
@@ -121,6 +121,12 @@ body {
     background-color: #FFCB00;
 }
 
+/* BEGIN Table of Contents */
+li.flc-toc-levels-items1, li.flc-toc-levels-items2, li.flc-toc-levels-items3, li.flc-toc-levels-items4 {
+    margin-top: 8px;
+    margin-bottom: 8px;
+}
```
Comment (jobara) [inline css/style.css]: "flc-" prefixed classes are only intended to be used for JS selectors. If possible make use of a different CSS selector for styling.

**Comment by jobara** [comment]
@jhung could you weigh in on the changes.

**Comment by jobara** [comment]
@sachin10101998 please also make sure you try with other adjustments such as font size, font style, enhance inputs, and all changes applied at the same time.

File: css/style.css (hunk 1 above)
Comment (sachin10101998) [reply css/style.css]: The classes for ul and li tags have been pre-defined in this manner. Should I rename the classes for these tags in the HTML code to some other suitable thing to make sure that "flc"- prefix classes are used only for JS selectors?
Screenshot from 2020-02-28 04-58-43

Comment (sachin10101998) [reply css/style.css]: The ul and li tags have "flc" prefixed class names as shown. Should I rename the classes to something more suitable to make sure the "flc" prefixed classes are only used for JS selectors?
Screenshot from 2020-02-28 06-53-00

**Comment by sachin10101998** [comment]
@jobara , the committed changes are compatible with all other combinations of font size, font style, enhanced inputs, etc.

File: css/style.css (hunk 1 above)
Comment (sachin10101998) [reply css/style.css]: fixed it with f6ec97c53a1f2dec18acb3024bbdb738b4c57030 .

**Comment by aashimgarg** [comment]
@jobara i think the best solution would be to keep the minimum line space to 1 as the table of contents does not collide at that point and by adding the margins we are only increasing the line space which is not required solution for this particular issue.
What are your views?

**Comment by sachin10101998** [comment]
@aashimgarg I already proposed that solution first in #110 and @jobara had the following views about it.
Screenshot from 2020-02-29 03-37-33

**Comment by sachin10101998** [comment]
@aashimgarg the major goal is not to keep a check on the absolute values of line spacing or any other UI component but adjusting the presentation of the content to something that suits the needs and preferences of the user.

**Comment by sachin10101998** [comment]
d6af681897e542b365e43af157a474ff33db3bb6 solves #108 too.

**Comment by greatislander** [comment]
Hi everyone, I'm going to close this as the redesign will likely invalidate the work here. Thanks for your contributions!

**Pull request closed without merging**